src/anim/controller/anim-blend-tree.js

```diff
@@ -204,6 +94,176 @@ class AnimBlendTree extends AnimNode {
         }
         return count;
     }
+
+    static between(num, a, b, inclusive) {
+        var min = Math.min(a, b),
+            max = Math.max(a, b);
+        return inclusive ? num >= min && num <= max : num > min && num < max;
+    }
+
+    static getAngleRad(a, b) {
+        return Math.atan2(a.x * b.y - a.y * b.x, a.x * b.x + a.y * b.y);
+    }
+
+    static clamp(num, min, max) {
+        return num <= min ? min : num >= max ? max : num;
+    }
+
+}
+
+/**
+ * @private
+ * @class
+ * @name AnimBlendTree1D
+ * @classdesc An AnimBlendTree that calculates it's weights using a 1D algorithm
+ * @description Create a new BlendTree1D.
+ */
+class AnimBlendTree1D extends AnimBlendTree {
+    calculateWeights() {
+        var i;
+        var parameterValue = this._findParameter(this._parameters[0]).value;
+        if (this.parametersEqual([parameterValue])) return;
+        this._parameterValues = [parameterValue];
+        this._children[0].weight = 0.0;
+        for (i = 0; i < this._children.length - 1; i++) {
+            var c1 = this._children[i];
+            var c2 = this._children[i + 1];
+            if (AnimBlendTree.between(parameterValue, c1.point, c2.point, true)) {
+                var child2Distance = Math.abs(c1.point - c2.point);
+                var parameterDistance = Math.abs(c1.point - parameterValue);
+                var weight = (child2Distance - parameterDistance) / child2Distance;
+                c1.weight = weight;
+                c2.weight = (1.0 - weight);
+            } else {
+                c2.weight = 0.0;
+            }
+        }
+    }
+}
+
+/**
+ * @private
+ * @class
+ * @name AnimBlendTreeCartesian2D
+ * @classdesc An AnimBlendTree that calculates it's weights using a 2D catesian algorithm
+ * @description Create a new BlendTree1D.
+ */
+class AnimBlendTreeCartesian2D extends AnimBlendTree {
+    pointDistanceCache(i, j) {
+        if (!this._pointCache) this._pointCache = {};
+        var pointKey = `${i}${j}`;
+        if (!this._pointCache[pointKey]) {
+            this._pointCache[pointKey] = this._children[j].point.clone().sub(this._children[i].point);
+        }
+        return this._pointCache[pointKey];
+    }
+
+    calculateWeights() {
+        var i, j, pi, parameterValues, minj, result, weightSum;
+        parameterValues = this.currentParameterValues();
+        if (this.parametersEqual(parameterValues)) return;
```

Comment:
I would probably combine these two lines to a single function. It would take this_parameterValues as parameter, it would update it .. and inside the same loop it would check each value for equal and return true or false.

Reply:
Makes sense, i've updated this.